core/core/util.go

```diff
@@ -0,0 +1,95 @@
+package xchaincore
```

Comment:
I think it could be better that you move this files of `Msg Event` into an independent package.